sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala

```diff
@@ -174,9 +174,15 @@ class ParquetToSparkSchemaConverter(
    */
   def convertField(
       field: ColumnIO,
-      sparkReadType: Option[DataType] = None): ParquetColumn = field match {
-    case primitiveColumn: PrimitiveColumnIO => convertPrimitiveField(primitiveColumn, sparkReadType)
-    case groupColumn: GroupColumnIO => convertGroupField(groupColumn, sparkReadType)
+      sparkReadType: Option[DataType] = None): ParquetColumn = {
+    val targetType = sparkReadType.map {
```

Comment:
I think there is another edge case in `convertInternal` when `fieldReadType` is `ArrayType` too - maybe we also need to check if the field type is UDT whose `sqlType` is `ArrayType`.

Reply:
I added TestArrayUDT type which represents Array[Long] internally in ParquetQuerySuite.